### PR TITLE
Friend List Part 6 - Friend List Complete/UI Bugs Fixed

### PR DESCRIPTION
**Pull request by olivercarson-pubnub**
PR for [DEVX-3359](https://pubnub.atlassian.net/browse/DEVX-3359) and [DEVX-3564](https://pubnub.atlassian.net/browse/DEVX-3564).

NOTE: In order to allow for Delete History, you'll need to replace the "SECRET_KEY_HERE" call made in Connector.cs with our app's secret key found in the admin portal. I'll share it in the group chat.

- Fully implemented request system. Friends can accept and reject friend requests. Friend requests are pulled from storage in case the user is offline or doesnt accept/deny the request.
- Chat (of all kinds) now simply displays for the user without being ran through profanity and translation
- Friend Chat filters for both profanity and translation when necessary
- Updated UI for Friend List and other social features of app.

### CODE REVIEW

File: Assets/SuperMultiplayerShooter/Scripts/FriendsList/FriendsList.cs [hunk 1]
```diff
@@ -223,6 +239,40 @@ public async Task<bool> PopulateFriendList(string channelGroup)
                     friendItem.Set(id, PNManager.pubnubInstance.CachedPlayers[id].Name);
                 }                  
             }
+
+            //Obtain history of messages from the friend request channel to determine if there are any pending invites.
+            PNResult<PNFetchHistoryResult> fetchHistoryResponse = await Connector.instance.GetPubNubObject().FetchHistory()
+                .Channels(new string[] { PubNubUtilities.chanFriendRequest + Connector.instance.GetPubNubObject().GetCurrentUserId() })
+                .ExecuteAsync();
+           if(fetchHistoryResponse != null && fetchHistoryResponse.Result != null && fetchHistoryResponse.Result.Messages != null && !fetchHistoryResponse.Status.Error)
```
Comment (darryncampbell-pubnub) [inline Assets/SuperMultiplayerShooter/Scripts/FriendsList/FriendsList.cs:247]: I thought by default the history API only returned something like 20 messages, but the documentation says 100.  I still think it is worth adding a comment that it's possible the API will only return SOME of the messages in the history (however unlikely that might be).  The documentation doesn't seem to say how you should know if there were additional messages which the API did not return, I thought there would be some kind of `more` flag but maybe I'm not understanding.

File: Assets/SuperMultiplayerShooter/Scripts/PNManager.cs [hunk 2]
```diff
@@ -307,6 +309,33 @@ public async Task<bool> DeleteChannelGroup(string channelGroup)
         return false;
     }
 
+    /// <summary>
+    /// Deletes all of the messages from the provided channel.
+    /// Note: This requires three features to be fully functional:
+    /// 1. Message Persistence is enabled for your key
+    /// 2. There is a setting to accept delete from history requests for a key, which you must enable by checking the Enable Delete-From-History checkbox in the key settings for your key in the Admin Portal.
+    /// 3. Requires Initialization with secret key (obtained pubnfrom the admin Portal..
```
Comment (darryncampbell-pubnub) [inline Assets/SuperMultiplayerShooter/Scripts/PNManager.cs:317]: I'll raise this on Slack with the Unity devs

File: Assets/SuperMultiplayerShooter/Scripts/SettingsWindow.cs [hunk 3]
```diff
@@ -111,12 +112,14 @@ public void SaveOptions()
                 }
                 
                 //Acceptable to not need to await for this call to finish.
```
Comment (darryncampbell-pubnub) [inline Assets/SuperMultiplayerShooter/Scripts/SettingsWindow.cs:114]: Need to delete this comment?

Comment (olivercarson-pubnub) [reply Assets/SuperMultiplayerShooter/Scripts/SettingsWindow.cs:114]: Will delete, thanks for the catch.

**Review by darryncampbell-pubnub — APPROVED** [review]

**Pull request merged**